tst/EnergyPlus/unit/OutputReportTabular.unit.cc

```diff
@@ -7653,3 +7654,206 @@ TEST_F(SQLiteFixture, WriteSourceEnergyEndUseSummary_TestPerArea) {
+    // General
+    EXPECT_NEAR(extLitUse * 6, gatherEndUseSubBEPS(1, DataGlobalConstants::endUseExteriorLights, 1), 1.);
+    // AnotherEndUseSubCat
+    EXPECT_NEAR(extLitUse * 3, gatherEndUseSubBEPS(2, DataGlobalConstants::endUseExteriorLights, 1), 1.);
+
+    OutputReportTabular::WriteBEPSTable();
+    OutputReportTabular::WriteDemandEndUseSummary();
+
+    EnergyPlus::sqlite->sqliteCommit();
+
+    // We test for Heating and Total, since they should be the same
+    std::vector<std::string> testReportNames = {"AnnualBuildingUtilityPerformanceSummary", "DemandEndUseComponentsSummary"};
+
+    std::string endUseName = "Exterior Lighting";
+    std::string endUseSubCategoryName = "AnotherEndUseSubCat";
+
+    for (auto& reportName: testReportNames) {
+
+        std::string query("SELECT RowName From TabularDataWithStrings"
+                          "  WHERE TableName = 'End Uses By Subcategory'"
+                          "  AND ColumnName = 'Subcategory'"
+                          "  AND ReportName = '" + reportName + "'"
+                          "  AND Value = '" + endUseSubCategoryName + "'");
+
+        auto result = queryResult(query, "TabularDataWithStrings");
+
+        ASSERT_EQ(1ul, result.size()) << "Query crashed for reportName=" << reportName;
+
+        // Add informative message if failed
+        EXPECT_EQ(endUseName, result[0][0]) << "Failed for reportName=" << reportName;
+    }
+
+    // Here's a demonstration of how the new format of the table could help in querying for a specific end use only, which was impossible before
+    // Here I'm returning ALL entries that belong to a specific end use (one row for each resource (fuel) type)
+
+    {
+        auto result = queryResult(
+                "SELECT * FROM TabularDataWithStrings"
+                "  WHERE TableName = 'End Uses By Subcategory'"
+                "  AND ReportName = 'AnnualBuildingUtilityPerformanceSummary'"
+                "  AND RowName = '" + endUseName + "'"
+                "  AND (TabularDataIndex - (SELECT TabularDataIndex FROM TabularDataWithStrings"
+                "                              WHERE TableName = 'End Uses By Subcategory'"
+                "                              AND ReportName = 'AnnualBuildingUtilityPerformanceSummary'"
+                "                              AND ColumnName = 'Subcategory'"
+                "                              AND RowName = '" + endUseName + "'"
+                "                              AND Value = '" + endUseSubCategoryName + "'))"
+                "       % (SELECT COUNT(Value) FROM TabularDataWithStrings"
+                "            WHERE TableName = 'End Uses By Subcategory'"
+                "            AND ReportName = 'AnnualBuildingUtilityPerformanceSummary'"
+                "            AND ColumnName = 'Subcategory') = 0",
+                "TabularDataWithStrings");
+        ASSERT_EQ(7u, result.size());
```

Comment:
This query works in one step to query all entries for a specific end use subcategory.